src/esm.ts

```diff
@@ -83,6 +86,10 @@ export type NodeLoaderHooksFormat =
   | 'module'
   | 'wasm';
 
+export type NodeImportAssertions = {
+  type: 'json' | 'wasm';
+};
```

Comment:
This is getting off topic, but while you both are available to discuss:

ts-node's resolver may need to become more complex in the future to support outDir->rootDir mappings and additional file extensions.  In that spirit, do you know if import assertions ever affect the resolution of a module specifier to a filesystem?  Wondering if it would be good to have typings for the `type` property or not.  Otherwise I like the opaque type suggestion.

Also, I don't see `type: 'wasm'` mention in node's docs anywhere.  Where did you find that?

Reply:
> Do you know if import assertions ever affect the resolution of a module specifier to a filesystem path? Wondering if it would be good to have typings for the `type` property or not. Otherwise I like the opaque type suggestion.

From [https://github.com/tc39/proposal-json-modules#motivation](JSON modules proposal)
> Rather than granting hosts free reign to implement JSON modules independently, specifying them in TC39 guarantees that they will behave consistently across all ECMA262-compliant hosts.

I guess It's up to the host implementation to handle the type assertions, but TC39 specifically wanted to specify the `type: "json"` so that the implementations would be compatible. If node or other hosts implement other types independently, the compatibility will be at greater risk.

> Also, I don't see `type: 'wasm'` mention in node's docs anywhere. Where did you find that?

True! I pre-empted them when I added the [`--experimental-wasm-modules`](https://nodejs.org/api/esm.html#wasm-modules) flag support, but that doesn't specify the import assertion at all now that I look at it. It's about just as likely that the import assertion type would end up being `webassembly`, so it might be better to drop it from the PR for now (or add both).